src/foam/nanos/auth/PermissionTableView.js

```diff
@@ -92,9 +95,19 @@ foam.CLASS({
                           .select(self.groupDAO.orderBy(self.Group.ID), function(g) {
                               var cb = foam.u2.md.CheckBox.create({data: self.checkPermissionForGroup(p.id, g)});
                               cb.data$.sub(function() { self.updateGroup(p, g, cb.data); });
-                              this.start('td').show(groups[g.id].data$.map(function() {return groups[g.id].data;}))
+                              this.start('td').show(groups[g.id].data$.map(function() { return groups[g.id].data; }))
                                     .style({'text-align': 'center', 'width': '100'}).tag(cb).call(function() {
-                                      if ( g.implies(p.id) ) { cb.style({'border-color': '#40C75B'}) };
+                                      self.groupDAO.find(g.parent).then(function(a) {
+                                        if ( a != undefined && a.implies(p.id) ) {
+                                          cb.setAttribute('title', g.parent + ': ' + p.id);
+                                          cb.style({'border-color': '#40C75B'})
+                                        }
+                                      });
+
+                                      if ( g.implies(p.id) ) {
```

Comment:
If g.implies(p.id) is true then you don't need to do the self.groupDAO.find() above, so move that code into an 'else' block of this if statement.